feat(iota-framework/stardust): make stardust models are generic

Dkwcs opened the pull request:
Stardust models have turned into generic
Rename iota variables to more general - "balance"

# Description of change

Stardust models(especially outputs) have generic balance field in order to apply not only IOTA coin.

## Links to any relevant issues

Fixes #577 .

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected) **possibly**? 

## How the change has been tested

Run existing unit tests for iota/crates/iota-framework/packages/stardust

Dkwcs committed:
feat(iota-framework/stardust):

Make outputs balance is generic
Rename iota variables to more general - "balance"

valeriyr reviewed (changes requested):
Hey @Dkwcs! Looks good in general!
I left several minor comments related to code simplification. Also, it is necessary to check and fix the documentation comments after changing IOTA to a generic type.

Dkwcs committed:
refactor(code comments, unit tests): Tiny refactoring of unit tests for alias, basic, nft to bring into the same view.
Iota mentions have been removed from comments, replacing with "coins/coin" term.

Dkwcs committed:
refactor(stardust/design-docs): change IOTA mention to more generic

lzpap reviewed (approved):
LGTM

kodemartin reviewed (approved):
LGTM

Dkwcs committed:
Lock file commit
